updated-knowbe4 (#1304)
SumoLogic/sumologic-documentation: docs/send-data/hosted-collectors/cloud-to-cloud-integration-framework/knowbe4-api-source.md

---
id: knowbe4-api-source
title: KnowBe4 API Source
sidebar_label: KnowBe4 API Source
description: This document explains how to configure the KnowBe4 Cloud-to-Cloud source setup using the Sumo logic environment.
---

import useBaseUrl from '@docusaurus/useBaseUrl';

The KnowBe4 API integration collects user data into Sumo Logic to store, analyze and alert. It ingests events data from the [Events API](https://developer.knowbe4.com/rest/userEvents#tag/Events/operation/listEvents).

:::note
To access the Knowbe4 API, you must have either the Platinum or Diamond user plan.
:::

## Prerequisites

> The KnowBe4 Source requires the following information to connect to KnowBe4 API:
> Before you begin setting up your **KnowBe4**** Source, the integration must be configured with the **Base URL** and **KnowBe4 API Token**.
> 1. Base URL
> 2. API Token

Reviewer: ```suggestion
Before you begin setting up your **KnowBe4** Source, which is required to connect to KnowBe4 API, you'll need to configured your integration with the **Base URL** and **KnowBe4 API Token**.
```

### Base URL

> The **Base URL** is the URL where your **KnowBe4** account is located  To get the base URL, follow the steps below:

Reviewer: ```suggestion
The **Base URL** is the URL where your **KnowBe4** account is located. To get the base URL, follow the steps below:
```

1. Log in to the **KnowBe4** application.
2. At the top of the browser, you will see the **Base URL** inside the address bar.
3. Choose the Base URL from the table below. The following table contains the base URLs based on the location of your **KnowBe4** account:

  | Server location | Server located at  | Base URLs |
  | :---|:---|:---|
  | US Server |	training.knowbe4.com | `https://api.events.knowbe4.com` |
  | EU Server | eu.knowbe4.com | `https://api-eu.events.knowbe4.com` |
  | CA Server |	ca.knowbe4.com | `https://api-ca.events.knowbe4.com` |

### API Token

The **API security token** is used to authenticate with KnowBe4 HTTP API. To get the **KnowBe4 API token**, follow the steps below:

> 1. Log into **KnowBe4** application.

Reviewer: ```suggestion
1. Log in to the **KnowBe4** application.
```

> 1. Navigate to **Events** section from left panel and click **Settings**.

Reviewer: ```suggestion
1. Navigate to the **Events** section from left panel and click **Settings**.
```

| ThirdPartyConfig  | Normally due to an invalid configuration. You'll need to review your Source configuration and make an update. | No retries are attempted until the Source is updated. | Not applicable                                                    | ThirdPartyConfigError  |
| ThirdPartyGeneric | Normally due to an error communicating with the third-party service APIs.                                     | Yes                                                   | The Source will retry for up to 90 minutes, after which it quits.                               | ThirdPartyGenericError |
| FirstPartyGeneric | Normally due to an error communicating with the internal Sumo Logic APIs.                                     | Yes                                                   | The Source will retry for up to 90 minutes, after which it quits.                               | FirstPartyGenericError |

### JSON configuration

Sources can be configured using UTF-8 encoded JSON files with the Collector Management API. See [how to use JSON to configure Sources](/docs/send-data/use-json-configure-sources) for details. 

| Parameter | Type | Required | Description | Access |

> |--|--|--|--|--|

Reviewer: ```suggestion
|:--|:--|:--|:--|:--|
```

| `config` | JSON Object  | Yes | Contains the [configuration-parameters](#config-parameters) of the Source. | not modifiable |
| `schemaRef` | JSON Object  | Yes | Use `{"type":"KnowBe4"}` for KnowBe4 Source. | not modifiable |
| `sourceType` | String | Yes | Use `Universal` for KnowBe4 Source. | not modifiable |

### Config Parameters

| Parameter | Type | Required | Description | Access |

> |---|---|---|---|---|

Reviewer: ```suggestion
|:---|:---|:---|:---|:---|
```
